pkg/mcp/sse.go

```diff
@@ -295,36 +294,12 @@ func (s *SSEClient) StartSession() (string, error) {
 		return "", fmt.Errorf("unexpected status code: %d", resp.StatusCode)
 	}
 
-	// Read the first SSE event which contains the message URL
-	scanner := bufio.NewScanner(resp.Body)
-	for scanner.Scan() {
-		line := scanner.Text()
-		if strings.HasPrefix(line, "endpoint: ") {
-			s.messageURL = strings.TrimSpace(strings.TrimPrefix(line, "endpoint: "))
-			break
-		}
-	}
-	if s.messageURL == "" {
-		resp.Body.Close()
-		return "", fmt.Errorf("failed to read message URL from SSE event")
-	}
-
-	// Parse the session ID from the message URL
-	parsedURL, err := url.Parse(s.messageURL)
-	if err != nil {
-		resp.Body.Close()
-		return "", fmt.Errorf("invalid message URL: %w", err)
-	}
+	session := make(chan sessionResponse)
 
-	sessID := parsedURL.Query().Get("sessionID")
-	if sessID == "" {
-		resp.Body.Close()
-		return "", fmt.Errorf("no session ID in message URL")
-	}
+	go s.listenMessages(resp.Body, session)
 
-	go s.listenMessages(sessID, resp.Body)
-
-	return sessID, nil
+	sessionResp := <-session
```

Comment:
This would hung the SSEClient, as it wait for initialize message from the server. In current implementation, the mcp.Client would send the initialize message to server after this call, as in MCP, the client should initiate the initialization. Also, listenMessage here is not only used for handling the initialize message, mcp.Client would also may receive [Roots](https://spec.modelcontextprotocol.io/specification/client/roots/) and [Sampling](https://spec.modelcontextprotocol.io/specification/client/sampling/) request from the server.

Reply:
FYI, I'm on my way to refactor the `Transport` interface, so you may wait to fix it in the upcoming design.

Reply:
I'm not sure that it would be blocked here – if you look in `listenMessages` we send the session back on that channel, so something will be received. When the server sends an `endpoint` message we parse that endpoint and send the session on this channel; we also send any error that occurs on this channel, to ensure that it never blocks.

Both the endpoint and the other messages are sent on the same stream. Previously, the code would use two `bufio.Scanner`s to parse this stream, which is buggy, as I explained in the PR; this code here isn't in any way different to the previous one other than using just one `bufio.Scanner` to parse it (inside `sse.Read`). Functionally it is the same.

The `session` channel basically receives the first event, the one with the endpoint. This receive here blocks until the first event is parsed, and not for as long as messages are read. Is there something I'm not seeing here?

Also, how can one test this code locally?

Reply:
It turn's out, the blocks happen because the SSEServer send the response in the old format (like you mentioned in the #1, so the `sse.Read` on `listenMessages` won't pick up the message. Let me commit the fix for SSEServer so it wont block.

To test this code locally, use:
`go test ./pkg/mcp/... -v -count=1`
I should put it somewhere on README, but before that, I think I need to remove that ugly `pkg` prefix.